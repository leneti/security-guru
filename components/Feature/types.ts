import type { ForwardRefExoticComponent, RefAttributes } from "react";
import type { Icon, IconProps } from "@tabler/icons-react";

export interface FeatureProps {
  icon: ForwardRefExoticComponent<Omit<IconProps, "ref"> & RefAttributes<Icon>>;
  title: React.ReactNode;
  description: React.ReactNode;
}
