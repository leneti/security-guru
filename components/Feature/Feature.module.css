.icon {
  color: var(--mantine-primary-color);
}
