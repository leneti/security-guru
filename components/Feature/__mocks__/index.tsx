import { FeatureProps } from "../types";

export function Feature({ title }: Partial<FeatureProps>) {
  return <div data-testid={title}>Mock Feature</div>;
}
