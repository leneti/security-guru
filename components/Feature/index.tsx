export { default as Feature } from "./Feature";
export * from "./types";
