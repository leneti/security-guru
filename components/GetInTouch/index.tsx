export { default, type ContactForm } from "./GetInTouch";
