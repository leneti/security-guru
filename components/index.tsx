export { default as ContactInfo } from "./ContactInfo";
export { default as Footer } from "./Footer";
export { default as GetInTouch, type ContactForm } from "./GetInTouch";
export { default as Header } from "./Header";
export { default as Hero } from "./Hero";
export { default as Logo } from "./Logo";
export { default as PageTitle } from "./PageTitle";
export { default as PageBackground, type PageBGProps } from "./PageBackground";
export { default as SlideDownTitle } from "./SlideDownTitle";
