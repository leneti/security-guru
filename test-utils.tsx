import type { ReactElement, ReactNode } from "react";
import { MantineProvider } from "@mantine/core";
import { render, type RenderOptions } from "@testing-library/react";
import { resolver, theme } from "@site/constants";

const AllTheProviders = ({ children }: { children: ReactNode }) => {
  return (
    <MantineProvider theme={theme} cssVariablesResolver={resolver}>
      {children}
    </MantineProvider>
  );
};

const customRender = (
  ui: ReactElement,
  options?: Omit<RenderOptions, "wrapper">,
) => render(ui, { wrapper: AllTheProviders, ...options });

export * from "@testing-library/react";
export { customRender as render };
