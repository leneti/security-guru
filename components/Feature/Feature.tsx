import { ThemeIcon, Title, useMantineTheme, Text } from "@mantine/core";
import { FeatureProps } from "@site/components/Feature";
import { getBGColor } from "@site/utils/getBGColor";

export default function Feature({
  icon: Icon,
  title,
  description,
}: FeatureProps) {
  const theme = useMantineTheme();

  return (
    <div>
      <ThemeIcon
        variant="filled"
        color={getBGColor(theme)}
        size={40}
        radius={40}
      >
        <Icon size="1.25rem" stroke={1.5} color={theme.fn.primaryColor()} />
      </ThemeIcon>
      <Title order={2} my="sm" variant="lightBg">
        {title}
      </Title>
      <Text size="sm" color="dimmed" sx={{ lineHeight: 1.6 }} variant="lightBg">
        {description}
      </Text>
    </div>
  );
}
