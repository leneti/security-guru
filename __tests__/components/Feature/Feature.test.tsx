import { Feature } from "@site/components/Feature";
import { render } from "@site/test-utils";

describe("Feature", () => {
  it("render", () => {
    const { container } = render(
      <Feature
        title="mock-feature"
        description="mock-description"
        icon={jest.fn(() => (
          <i>mock-icon</i>
        ))}
      />,
    );
    expect(container).toMatchSnapshot();
  });
});
