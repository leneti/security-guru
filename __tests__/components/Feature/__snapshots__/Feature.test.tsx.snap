// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Feature render 1`] = `
<div>
  <div>
    <div
      class="mantine-ThemeIcon-root mantine-1jywupq"
    >
      <i>
        mock-icon
      </i>
    </div>
    <h2
      class="mantine-Text-root mantine-Title-root mantine-1qt2cm2"
    >
      mock-feature
    </h2>
    <div
      class="mantine-Text-root mantine-1poidil"
    >
      mock-description
    </div>
  </div>
</div>
`;
